src/server/bg_services/namespace_monitor.js

```diff
@@ -46,96 +49,140 @@ class NamespaceMonitor {
                 }
                 dbg.log1(`test_namespace_resource_validity: namespace resource ${nsr.name} has error as expected`);
             }
         });
-        dbg.log1(`test_namespace_resource_validity finished successfuly..`);
+        dbg.log1(`test_namespace_resource_validity finished successfully..`);
     }
 
-    init_nsr_connection_to_target(nsr) {
+    check_error_codes(err, nsr) {
         if (nsr.nsfs_config) {
-            return;
-        }
-        const {endpoint, access_key, secret_key} = nsr.connection;
-        let conn;
-        switch (nsr.connection.endpoint_type) {
-            case 'AWS' || 'S3_COMPATIBLE' || 'IBM_COS': {
-                conn = new AWS.S3({
-                    endpoint: endpoint,
-                    credentials: {
-                        accessKeyId: access_key.unwrap(),
-                        secretAccessKey: secret_key.unwrap()
-                    },
-                    s3ForcePathStyle: true,
-                    sslEnabled: false
-                });
-                break;
-            }
-            case 'AZURE': {
-                const conn_string = cloud_utils.get_azure_connection_string({
-                    endpoint,
-                    access_key: access_key,
-                    secret_key: secret_key
-                });
-                conn = azure_storage.createBlobService(conn_string);
-                break;
-            }
-            default:
-                dbg.error('namespace_monitor: invalid endpoint type', nsr.endpoint_type);
+            return true;
+        } else if (err.code !== 'BlobNotFound' && err.code !== 'NoSuchKey') {
+            const { endpoint, target_bucket } = nsr.connection;
+            dbg.error('test_namespace_resource_validity failed:', err, endpoint, target_bucket);
+            return true;
         }
+        return false;
+    }
+
+    async run_update_issues_report(err, nsr) {
+        await this.client.pool.update_issues_report({
+            namespace_resource_id: nsr._id,
+            error_code: err.code,
+            time: Date.now(),
+            monitoring: true
+        }, {
+            auth_token: auth_server.make_auth_token({
+                system_id: system_store.data.systems[0]._id,
+                account_id: system_store.data.systems[0].owner._id,
+                role: 'admin'
+            })
+        });
+    }
+
+    init_nsr_s3_connection_to_target(nsr) {
+        const { endpoint, access_key, secret_key } = nsr.connection;
+        const conn = new AWS.S3({
+            endpoint: endpoint,
+            credentials: {
+                accessKeyId: access_key.unwrap(),
+                secretAccessKey: secret_key.unwrap()
+            },
+            s3ForcePathStyle: true,
+            sslEnabled: false
+        });
+
+        this.nsr_connections_obj[nsr._id] = conn;
+    }
+
+    init_nsr_blob_connection_to_target(nsr) {
```

Comment:
I would just embed this function content directly inside `test_blob_resource()`.